### ISSUE TITLE
[bug] Literal strings are matched generically

### ISSUE THREAD

**Issue by bruceg** [ast-grep/ast-grep#1057]
# ⏯ Playground Link

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiUGF0Y2giLCJsYW5nIjoicnVzdCIsInF1ZXJ5IjoiJEUud2l0aF9tZWFuaW5nKCRELCBcIm1lYW5pbmdcIikiLCJyZXdyaXRlIjoiJEUud2l0aF9tZWFuaW5nKCRELCBtZWFuaW5nOjpNRVNTQUdFKSIsImNvbmZpZyI6IiMgWUFNTCBSdWxlIGlzIG1vcmUgcG93ZXJmdWwhXG4jIGh0dHBzOi8vYXN0LWdyZXAuZ2l0aHViLmlvL2d1aWRlL3J1bGUtY29uZmlnLmh0bWwjcnVsZVxucnVsZTpcbiAgYW55OlxuICAgIC0gcGF0dGVybjogY29uc29sZS5sb2coJEEpXG4gICAgLSBwYXR0ZXJuOiBjb25zb2xlLmRlYnVnKCRBKVxuZml4OlxuICBsb2dnZXIubG9nKCRBKSIsInNvdXJjZSI6ImZuIHRlc3QoKSB7XG4gICAgc2NoZW1hLndpdGhfbWVhbmluZyhGT08sIFwibWVhbmluZ1wiKTtcbiAgICBzY2hlbWEud2l0aF9tZWFuaW5nKEZPTywgXCJzZXJ2aWNlXCIpO1xufSJ9

# 💻 Code

```rust
fn test() {
    schema.with_meaning(FOO, "meaning");
    schema.with_meaning(FOO, "service");
}
```

Pattern:

```yml
$E.with_meaning($D, "meaning")
```

# 🙁 Actual behavior

The pattern matches both of the calls in the above code snippet, with `"meaning"` matching `"service"` in the second.

# 🙂 Expected behavior

I expected that it would only match the first one, since the constant literal string is different from the pattern. My use case is to replace a literal string with a predefined constant, but the pattern is matching the wrong strings to be useful.

**Comment by HerringtonDarkholme** [comment]
tree-sitter does not report string literal's content. 

<img width="378" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/2be54d01-06cb-405f-b527-83c53fcb3f26">

**Comment by HerringtonDarkholme** [comment]
This is fixed in https://github.com/tree-sitter/tree-sitter-rust/issues/218

But it is blocked by https://github.com/tree-sitter/tree-sitter-typescript/blob/master/Cargo.toml#L29

**Comment by HerringtonDarkholme** [comment]
You can use this workaround

```yaml
rule:
  pattern: $E.with_meaning($D, $STR)
constraints:
  STR: { regex: '^"meaning"$'}
```

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6InJ1c3QiLCJxdWVyeSI6IiRFLndpdGhfbWVhbmluZygkRCwgXCJtZWFuaW5nXCIpIiwicmV3cml0ZSI6IiRFLndpdGhfbWVhbmluZygkRCwgbWVhbmluZzo6TUVTU0FHRSkiLCJjb25maWciOiIjIFlBTUwgUnVsZSBpcyBtb3JlIHBvd2VyZnVsIVxuIyBodHRwczovL2FzdC1ncmVwLmdpdGh1Yi5pby9ndWlkZS9ydWxlLWNvbmZpZy5odG1sI3J1bGVcbnJ1bGU6XG4gIHBhdHRlcm46ICRFLndpdGhfbWVhbmluZygkRCwgJFNUUilcbmZpeDpcbiAgbG9nZ2VyLmxvZygkQSlcbmNvbnN0cmFpbnRzOlxuICBTVFI6IHsgcmVnZXg6ICdeXCJtZWFuaW5nXCIkJ30iLCJzb3VyY2UiOiJmbiB0ZXN0KCkge1xuICAgIHNjaGVtYS53aXRoX21lYW5pbmcoRk9PLCBcIm1lYW5pbmdcIik7XG4gICAgc2NoZW1hLndpdGhfbWVhbmluZyhGT08sIFwic2VydmljZVwiKTtcbn0ifQ==

**Comment by HerringtonDarkholme** [comment]
I have created a new issue to track the fix. Closing this now.
